make_model_order.py

```diff
@@ -0,0 +1,110 @@
+# create VAR model order
+import numpy as np
+from scipy import linalg
+import math
+def cov(X, p):
+    """vector autocovariance up to order p
+
+    Parameters
+    ----------
+    X : ndarray, shape (N, n)
+        The N time series of length n
+
+    Returns
+    -------
+    R : ndarray, shape (p + 1, N, N)
+        The autocovariance up to order p
+    """
+    N, n = X.shape
+    R = np.zeros((p + 1, N, N))
+    for k in range(p + 1):
+        R[k] = (1. / float(n - k)) * np.dot(X[:, :n - k], X[:, k:].T)
+    return R
+    
+def mvar_fit(X, p):
+    """Fit MVAR model of order p using Yule Walker
+
+    Parameters
+    ----------
+    X : ndarray, shape (N, n)
+        The N time series of length n
+    n_fft : int
+        The length of the FFT
+
+    Returns
+    -------
+    A : ndarray, shape (p, N, N)
+        The AR coefficients where N is the number of signals
+        and p the order of the model.
+    sigma : array, shape (N,)
+        The noise for each time series
+    """
+    N, n = X.shape
+    gamma = cov(X, p)  # gamma(r,i,j) cov between X_i(0) et X_j(r)
+    G = np.zeros((p * N, p * N))
+    gamma2 = np.concatenate(gamma, axis=0)
+    gamma2[:N, :N] /= 2.
+
+    for i in range(p):
+        G[N * i:, N * i:N * (i + 1)] = gamma2[:N * (p - i)]
+
+    G = G + G.T  # big block matrix
+
+    gamma4 = np.concatenate(gamma[1:], axis=0)
+
+    phi = linalg.solve(G, gamma4)  # solve Yule Walker
+
+    tmp = np.dot(gamma4[:N * p].T, phi)
+    sigma = gamma[0] - tmp - tmp.T + np.dot(phi.T, np.dot(G, phi))
+
+    phi = np.reshape(phi, (p, N, N))
+    for k in range(p):
+        phi[k] = phi[k].T
+
+    return phi, sigma
+    
+def compute_order(X, p_max):
+    """Estimate AR order with BIC
+
+    Parameters
+    ----------
+    X : ndarray, shape (N, n)
+        The N time series of length n
+    p_max : int
+        The maximum model order to test
+
+    Returns
+    -------
+    p : int
+        Estimated order
+    bic : ndarray, shape (p_max + 1,)
+        The BIC for the orders from 0 to p_max.
+    """
+    N, n = X.shape
+
+    bic = np.empty(p_max + 1)
+    bic[0] = np.inf # XXX
+
+    Y = X.T
+
+    for p in range(1, p_max + 1):
+        print p
+        A, sigma = mvar_fit(X, p)
+        A_2d = np.concatenate(A, axis=1)
+
+        n_samples = n - p
+        bic[p] = n_samples * N * math.log(2. * math.pi)
+        bic[p] += n_samples * np.log(linalg.det(sigma))
+        bic[p] += p * (N ** 2) * math.log(n_samples)
+
+        sigma_inv = linalg.inv(sigma)
+        S = 0.
+        for i in range(p, n):
+            res = Y[i] - np.dot(A_2d, Y[i - p:i][::-1, :].ravel())
+            S += np.dot(res, sigma_inv.dot(res))
+
+        bic[p] += S
+
+    p = np.argmin(bic)
+    return p, bic
```

Comment:
In fact, I think all three of these functions can be added to jumeg_math module.